app/controllers/additional_details/submissions_controller.rb

```diff
@@ -44,6 +44,8 @@ def create
     @submissions = @protocol.submissions
     @line_item = LineItem.find(submission_params[:line_item_id])
     @service_request = @line_item.service_request
+    @permission_to_edit = @protocol.project_roles.where(identity: current_user, project_rights: ['approve', 'request']).any?
```

Comment:
This logic is already in the Identity model as `has_correct_project_role?`, so `@permission_to_edit = current_user.has_correct_project_role?(@service_request)` would be better here.